src/js/server/hcapi/applications.js

```diff
@@ -1,37 +1,163 @@
 const { Router } = require('express');
-const { Hacker, HackerApplication, ApplicationResponse } = require('js/server/models');
+const Sequelize = require('sequelize');
+const { Hacker, HackerApplication, ApplicationResponse, TeamMember, Team, db } = require('js/server/models');
 const { createHttpError } = require('./errors');
+const fs = require('fs');
+const individualScoreQuery = fs.readFileSync('src/js/server/hcapi/queries/indiv_score_unteamed.sql', 'utf8');
 
 const applicationsRouter = new Router();
 
+// Takes an application and the set of all individual scores and
+// the set of teams average scores
+// returns the score for that application as a float
+function calculateScore(application, individualScores, teamScores) {
+  let hacker = application.hacker;
+  let teamId = hacker.Team ? hacker.Team.teamId : null;
+
+  if (teamId == null) {
+    // The hacker isn't in a team
+    if (application.inTeam) {
+      // If they applied as a team then their application is incomplete, return null
+      return null
+    } else {
+      // return the individual average score if it exists, null otherwise
+      let score = individualScores[application.id];
+      return score ? score : null;
+    }
+  } else {
+    // The hacker is in a team 
+    let teamScore = teamScores[teamId];
+    if (teamScore === undefined) {
+      // Something went wrong, couldn't find the hackers team
+      console.log(`Error: Could not find score for team ${teamId}`);
```

Comment:
Whoops! Had intended to return null here, must have gotten lost somewhere along the way. We could throw an error, yes. I'm unsure what's the best behaviour here. This should never happen unless something goes wrong in Sequelize whereby it returns a Hacker-Team relation but cannot find the team itself. What are your thoughts?